src/model/network-simple-multicast.hpp

```diff
@@ -161,7 +161,7 @@ class SimpleMulticastNetwork : public Network
   void SetTileWidth(double width_um);
 
   // Parse ERT to get multi-casting energy
-  double GetMulticastEnergy(std::uint64_t multicast_factor);
+  double GetMulticastEnergy(std::uint64_t multicast_factor, std::string data_space_name);
  
```

Comment:
I understand what you are trying to achieve here. But at the same time, this class is probably not a SimpleMulticast network anymore... And this class also becomes highly dependent on using Accelergy in a particular way. In the future, one would probably like to use SimpleMulcastNetwork without customized Accelergy ERT. 

Several suggestions we can consider:
1. Create two separated functions GetMulticastEnergy() and GetMulticastEnergyByDataType(). Configure the network for what to use with attributes. 
2. Avoid hard-coded, user-defined action name ("transfer_") in the source code. Maybe make it a attribute of the SimpleMulticastNetwork too.
3. Create yet another network class that inherited all basic features of a multicast network but supports all Accelergy-centric features.